.github/workflows/check.yaml

```diff
@@ -1,140 +1,169 @@
+      matrix:
+        runs-on: [[ "self-hosted", "jammy", "amd64", "two-xlarge" ]]
+        test-command: [ "TEST_MODEL_SETTINGS='update-status-hook-interval=30s' tox -e func" ]
+        juju-channel: [ "3.4/stable" ]
     steps:
+
       - uses: actions/checkout@v4
         with:
           submodules: true
 
+        # arm64 runners don't have gcc installed by default
+      - name: Install dependencies
+        run: |
+          sudo apt update
+          sudo apt install -y gcc
+
       - name: Setup Python
         uses: actions/setup-python@v5
         with:
           python-version: "3.10"
 
-      - name: Install tox latest release
-        run: |
-          python -m pip install --upgrade pip
-          python -m pip install tox
-
-      - name: external controller juju config
-        run: |
-          mkdir -p $HOME/.local/share/juju
-          # The credential are stored in repository's secret as base64 encoded string.
-          # This help to avoid special characters that can interface with the parsing of YAML/JSON files.
-          echo ${{ secrets.JUJU_CONTROLLERS_YAML }} | base64 -d > $HOME/.local/share/juju/controllers.yaml
-          echo ${{ secrets.JUJU_ACCOUNTS_YAML }} | base64 -d > $HOME/.local/share/juju/accounts.yaml
-
-      - name: external controller openstack environment
-        run: |
-           # Workaround because github workflow doesn't allow to use secrets key
-           # word in an if conditional expression.
-           if [ "${{ secrets.OPENSTACK_AUTH_ENV }}" != '' ]; then
-             # Add mask to secret
-             # https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#example-masking-a-generated-output-within-a-single-job
-             echo ${{ secrets.OPENSTACK_AUTH_ENV }} \
-                | base64 -d \
-                | grep -E 'OS_AUTH_URL|OS_USERNAME|OS_PASSWORD|OS_PROJECT_NAME|OS_PROJECT_ID' \
-                | awk -F'=' '{print $2}' \
-                | xargs -I {} echo "::add-mask::{}"
-             echo ${{ secrets.OPENSTACK_AUTH_ENV }} | base64 -d >> "$GITHUB_ENV"
-           else
-             echo "Skip because missing openstack-auth-env secrets"
-           fi
-
-      - name: append to zaza yaml
-        run: |
-           cat >> $HOME/.zaza.yaml <<EOF
-           ---
-           model_settings:
-             image-stream: released
-             update-status-hook-interval: 30s
-           region: prodstack6
-           cloud: builder-cloud
-           credential: builder-cloud-cred
-           EOF
-
-      - name: Configure latest lxd (to build the charm in)
-        uses: canonical/setup-lxd@2aa6f7caf7d1484298a64192f7f63a6684e648a4  # latest main at time of writing
+      - name: Setup Juju environment
+        uses: charmed-kubernetes/actions-operator@main
         with:
-          channel: latest/stable
+          provider: "lxd"
+          juju-channel: ${{ matrix.juju-channel }}
+          charmcraft-channel: "2.x/stable"
```

Comment:
nit: cloudsupport supports charmcraft 3 - we were previously using charmcraft3 in the workflow here: https://github.com/canonical/charm-cloudsupport/blob/0c2a782c97e45d5555145f5b18a240ea32b32afd/.github/workflows/check.yaml#L116-L116